Install Page Update

nsryan2 opened the pull request:
This PR updates the installation page of the website with the information @mehmeturkmen and I put together. It adds information about the Ubuntu scripts, adds a link to the How to test guide, and removes the "Create a Package (For Developer)" instructions on the conda install page (conda-recipes has been depreciated).

nsryan2 committed:
Merge pull request #5 from pyne/0.7.0-rc

Update my fork

nsryan2 committed:
Add more information about the files and use

This commit adds information based off of the readme in the scripts repository.

nsryan2 committed:
Add a link to How to test

This PR adds a link from the install page to the page on how to test the installation of PyNE.

nsryan2 committed:
Remove info on installing with conda-recipes

Conda-recipes has been depreciated and is no longer maintained, as such the version of PyNE that it has is out of date with no hope of being updated.

nsryan2 commented:
~~Sorry about the messy history, I accidentally compromised my default and had to make a backup.~~

Edit: Only the last three commits are relavent to this PR

nsryan2 committed:
Update the changelog

kkiesling commented on docs/install/ubuntu.rst:
```diff
@@ -7,3 +7,49 @@ Ubuntu build script
+Example for building a docker image of the latest stable branch of PyNE based on
+Ubuntu 16.04::
+
+    docker build -f ubuntu_16.04-stable.dockerfile -t pyne-16.04-stable .
```
```suggestion
    docker build -f ubuntu_16.04-stable.dockerfile -t pyne-ubuntu-16.04-stable .
```

kkiesling commented on docs/install/ubuntu.rst:
```diff
@@ -7,3 +7,49 @@ Ubuntu build script
+
+The intention of these
+scripts is that PyNE can be ready to use on a clean install of any of
+the supported operating systems. Furthermore, the user should choose either
```
Can we link "supported operating systems" to the page that lists this? (or list them here if there is no page)

nsryan2 replied:
Yes we could, however that page still needs to be updated. I'll add those updates to this PR, and then link it here

gonuke commented on docs/install/ubuntu.rst:
```diff
@@ -7,3 +7,49 @@ Ubuntu build script
 Scripts for installing PyNE and all its dependencies from scratch on Ubuntu
 14.04 - 15.04 are found `here
```
```suggestion
16.04 & 18.04 are found `here
```
Is anyone game for adding 20.04?

gonuke reviewed (changes requested):
We need to remove references to Windows from the `index.rst`.

Also a philosophical question: Should the instructions for using the build scripts be in the README of the install_scripts repo and then we just link to there?  Pro: PR's to install_scripts can be reviewed for updates to those instructions to ensure consistency. Con: they will not be formatted nicely as part of the website.

nsryan2 committed:
Update docs/install/ubuntu.rst

Co-authored-by: Kalin Kiesling <krkiesling@gmail.com>

nsryan2 committed:
Update docs/install/ubuntu.rst

Co-authored-by: Paul Wilson <paul.wilson@wisc.edu>

nsryan2 commented:
> Also a philosophical question: Should the instructions for using the build scripts be in the README of the install_scripts repo and then we just link to there? Pro: PR's to install_scripts can be reviewed for updates to those instructions to ensure consistency. Con: they will not be formatted nicely as part of the website.

After thinking about it some more, it would probably be best to just link to the readme here. This way, if someone went straight to the install scripts they wouldn't miss the instructions or have to worry about differences between two sets of instructions.

nsryan2 committed:
Update the platform info

I'm not sure if this is enough information, and I'm a little concerned that it might give someone the wrong opinion of their ability to install and use PyNE

nsryan2 committed:
Add "use" for some more clarity

nsryan2 committed:
Add link to supported os

nsryan2 committed:
Remove instructions to consolidate documentation

bam241 commented:
this need to be rebased and pushed

opotowsky commented:
Hey @nsryan2 It has been brought to my attention that [openMC](https://docs.openmc.org/en/stable/index.html) is an optional dependency listed nowhere that I can see. 

Could you add it to optional dependencies on `docs/install/linux_source.rst` and the main pyne `readme.rst`?

opotowsky commented:
> Hey @nsryan2 It has been brought to my attention that [openMC](https://docs.openmc.org/en/stable/index.html) is an optional dependency listed nowhere that I can see.
> 
> Could you add it to optional dependencies on `docs/install/linux_source.rst` and the main pyne `readme.rst`?

This would also maybe affect the install scripts, but someone else can speak to that (I think some do install the optional dependencies)

bam241 commented:
> > Hey @nsryan2 It has been brought to my attention that [openMC](https://docs.openmc.org/en/stable/index.html) is an optional dependency listed nowhere that I can see.
> > Could you add it to optional dependencies on `docs/install/linux_source.rst` and the main pyne `readme.rst`?
> 
> This would also maybe affect the install scripts, but someone else can speak to that (I think some do install the optional dependencies)

we don't support openMC in the install script for now...

nsryan2 commented:
> we don't support openMC in the install script for now...

@opotowsky Is openMC for the testing? If so, should I just add a note on the how to test pyne page that you need it to run the tests?

kkiesling commented:
> > we don't support openMC in the install script for now...
> 
> Is openMC for the testing? If so, should I just add a note on the how to test pyne page that you need it to run the tests?

If you don't have OpenMC the tests should just be skipped I think? I don't think OpenMC needs to be listed in these files if I am processing what this PR is for. I think it just needs to be added to the dependency list as an optional dependency just like MOAB and DAGMC.

kkiesling commented on docs/install/linux_source.rst:
```diff
@@ -25,6 +25,7 @@ PyNE has the following dependencies:
 Optional Depenendencies:
    #. `MOAB <https://press3.mcs.anl.gov/sigma/moab-library>`_
    #. `PyTAPS <https://pythonhosted.org/PyTAPS/index.html>`_
```
PyTAPS needs removed here and DAGMC needs to be added (we should probably note that DAGMC requires MOAB?)

nsryan2 replied:
The link that I provided for DAGMC takes you to the page that encompasses both the link to the dependencies and the link to the installation instructions, which I think should do the trick. If I'm not mistaken, we have other dependencies that rely on each other in this way and we don't specify for those.

kkiesling replied:
ya true- probably not necessary to note here then

kkiesling commented on readme.rst:
```diff
@@ -41,6 +41,7 @@ PyNE has the following dependencies:
 Optional Depenendencies:
    #. `MOAB <https://press3.mcs.anl.gov/sigma/moab-library>`_
    #. `PyTAPS <https://pythonhosted.org/PyTAPS/index.html>`_
```
same thing- PyTAPS to be removed, DAGMC needs to be added

kkiesling reviewed (changes requested):
It won't let me add specific line comments on non changed lines, but in the `readme.rst` there needs to be a few more changes:

* line 66 says `where VERSION should be replaced with the version number to be installed.` but there is no `VERSION` in the command
* the next paragraph talks about a windows binary but we don't support that so we should just remove that entire paragraph.

mehmeturkmen commented on docs/install/supported.rst:
```diff
@@ -4,22 +4,12 @@
- * OSX 10.8 - x86_64
+ * Ubuntu 16.04 - x86_64
+ * Ubuntu 18.04 - x86_64
 
```
We can also add Ubuntu 20.04 - x86_64 here as a supported platform. I installed pyNE from source and ubuntu_18.04.sh file. It is working with no problem.

nsryan2 committed:
Merge branch '0.7.0-rc' into install_page_update
